File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {
        int slow = 0;
        int fast = 1;
        int count = 1;

        while(fast < nums.Length){
            if(nums[fast] > nums[slow]){
                ++slow;
                nums[slow] = nums[fast];
                count = 1;
                ++fast;
                continue;
            }
            //found same number
            if(count < 2){
                //this is 2nd appearance
                ++slow;
                nums[slow] = nums[fast];
                ++count;
            }
            ++fast;
        }
        return slow+1;
    }
}

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        slow = 0
        fast = 1
        count = 1

        while fast < len(nums):
            if nums[fast] > nums[slow]:
                slow+=1
                nums[slow] = nums[fast]
                count = 1
                fast += 1
                continue
            #found same number
            if count < 2:   #2nd appearance
                slow+=1
                nums[slow] = nums[fast]
                count+=1
            fast +=1

        return slow+1